src/ToricVarieties.jl

```diff
@@ -21,38 +32,55 @@ function NormalToricVariety( rays::Matrix{Int}, cones::Vector{Vector{Int}} )
     variety = GAP.Globals.ToricVariety( fan )
 
     Incidence = Oscar.IncidenceMatrix(cones)
-    arr = @Polymake.convert_to Array{Set{Int}} Polymake.common.rows(Incidence.pm_incidencematrix)
+    arr = @Polymake.convert_to Array{Set{Int}} GAP.Globals.POLYMAKE_JULIA_MODULE.common.rows(Incidence.pm_incidencematrix)
 
-    pmntv = Polymake.fulton.NormalToricVariety(
+    pmntv = GAP.Globals.POLYMAKE_JULIA_MODULE.fulton.NormalToricVariety(
         RAYS = Oscar.matrix_for_polymake(rays),
         MAXIMAL_CONES = arr,
     )
 
     # wrap it into a struct and return
     return NormalToricVariety( variety, pmntv )
 end
+export NormalToricVariety
+
+"""
+    NormalToricVariety( v::GapObj )
 
+Construct the Julia wrapper for a `GAP` toric variety `v`.
+"""
 function NormalToricVariety(GapNTV::GapObj)
    pmNTV = ntv_gap2polymake(GapNTV)
    return NormalToricVariety(GapNTV, pmNTV)
 end
+export NormalToricVariety
+
+"""
+    ntv_gap2polymake( v::GapObj )
 
+Convert a `GAP` toric variety `v` into a `Polymake` toric variety.
+"""
 function ntv_gap2polymake(GapNTV::GapObj)
     ff = GAP.Globals.Fan(GapNTV)
     R = GAP.Globals.RayGenerators(ff)
     MC = GAP.Globals.RaysInMaximalCones(ff)
     rays = Matrix{Int}(R)
     cones = [findall(x->x!=0, vec) for vec in [[e for e in mc] for mc in MC]]
     Incidence = Oscar.IncidenceMatrix(cones)
-    arr = @Polymake.convert_to Array{Set{Int}} Polymake.common.rows(Incidence.pm_incidencematrix)
-    pmntv = Polymake.fulton.NormalToricVariety(
+    arr = @Polymake.convert_to Array{Set{Int}} GAP.Globals.POLYMAKE_JULIA_MODULE.common.rows(Incidence.pm_incidencematrix)
+    pmntv = GAP.Globals.POLYMAKE_JULIA_MODULE.fulton.NormalToricVariety(
```

Comment:
Why is `Polymake` replaced here by a GAP variable holding presumably the same value?

Reply:
Just to be clear: of course you can just merge this PR (this is up to you anyway), and this could be cleaned up later. But I am really confused why it is necessary; I hope it isn't (then it should be reverted), but if it is (i.e. you changed it because it gave you an error before), then I would really like to investigate it, as it sounds like there's a deeper bug and the above change is just plastering over it. Which one is it?

Reply:
This change is probably not absolutely necessary.

I was trying to follow one of your previous suggestions and defined `JToric.GAP.Globals.POLYMAKE_JULIA_MODULE = Polymake` in the `__init__`. My first attempt was to replace `Polymake` by `POLYMAKE_JULIA_MODULE`. If I recall correctly, this lead to an undef_var error. A call via

`JToric.GAP.Globals.POLYMAKE_JULIA_MODULE`

worked.

I then tried if possibly the leading `JToric` could be removed (global variable inside package `JToric`, so maybe this is automatically taken into account by `Julia`). Indeed, this worked and lead to the above code.

If this is serious, then I prefer to fix it before merging.

Reply:
I should say that this appears here, as a consequence of my confusion about the earlier failures of this PR. Maybe the cleanest way is indeed to revert this change and address it separately in a PR to https://github.com/oscar-system/JToric.jl/issues/40?